web-gateway/src/main/java/controllers/UserController.java

```diff
@@ -57,6 +57,20 @@ public UserController(Configuration config, MessagesApi messagesApi, UserService
         );
     }
 
+    public CompletionStage<Result> logoutUser() {
+        Http.Context ctx = ctx();
```

Comment:
I think this method can be reduced to just two lines and still do everything that this one does:

```java
ctx().session().clear();
return ok(views.html.logout.render(nav));
```

Reply:
@TimMoore I think `nav` would be unknown. There's an intermediate solution between your two options:

```java
    public CompletionStage<Result> logoutUser() {
        return loadNav(Optional.empty()).thenApply(nav -> {
            ctx().session().clear();
            return ok(views.html.logout.render(nav));
        });
    }
```